### PR TITLE
fix: decoding original purl if needed

### PR DESCRIPTION
**Pull request by olavtar**
fix: Implement URL decoding for original purl.

### CODE REVIEW

**Review by xieshenzh — APPROVED** [review]

File: ui/src/mocks/reportDocker.mock.ts [hunk 1]
```diff
@@ -843,7 +843,7 @@ export const dockerReport: AppData = {
         }
       }
     },
-    "pkg:oci/default-app@sha256:333224a233db31852ac1085c6cd702016ab8aaf54cecde5c4bed5451d636adcf?repository_url=quay.io/default-app&tag=0.0.1": {
+    "pkg:oci/alpine@sha256%3A1fafb0905264413501df60d90a92ca32df8a2011cbfb4876ddff5ceb20c8f165?arch=amd64&os=linux&repository_url=test.io%2Ftest%2Falpine&tag=test-version": {
```
Comment (ruromero) [inline ui/src/mocks/reportDocker.mock.ts:846]: Shouldn't this be done in the json report?

**Review by ruromero — CHANGES_REQUESTED** [review]
Isn't the json report affected as well?

**Comment by xieshenzh** [comment]
> Isn't the json report affected as well?

No. The purl in the json report is fine. It is ok to use URL-encoded purl in the json report. 
But when the purl is URL-encoded, link to the UBI image in the html report is not generated as expected. This PR fixes this issue.

**Comment by ruromero** [comment]
> No. The purl in the json report is fine. It is ok to use URL-encoded purl in the json report. But when the purl is URL-encoded, link to the UBI image in the html report is not generated as expected. This PR fixes this issue.

I don't think it's ok to use a URL-encoded purl in the json report. This will affect the html report `dependency` column and also the clients won't know if the PURL is encoded or not. We have to make sure the PURL is either always encoded or always unencoded. In all other cases it's not encoded.

**Comment by xieshenzh** [comment]
> I don't think it's ok to use a URL-encoded purl in the json report. This will affect the html report `dependency` column and also the clients won't know if the PURL is encoded or not. We have to make sure the PURL is either always encoded or always unencoded. In all other cases it's not encoded.

This PR deals with the keys in the request body for batch analysis. 
The exhort backend doesn't change the keys. 
If the keys are encoded in the request, they will still be encoded as the keys of the json reports in the response. 
If the keys are not encoded in the request, the keys in the response will be unencoded.

Then the json reports (i.e. dictionary) will be used to generate the html report. 
This PR handles the scenario when encoded keys (from the request body) are used to generate the html report.

In terms of the recommendation in the json report for UBI, they are always url-encoded.
The type of both `ref` and `recommendation` are `PackageRef`, so they are generated with the `packageurl-java` lib, which encodes the values.

**Comment by ruromero** [comment]
As discussed in the meeting, if it only affects the dictionary key for the batch-analysis then it's OK to only decode the pURL in the report.

**Review by ruromero — APPROVED** [review]

**Pull request merged**